sources/engine/Stride.Rendering/Rendering/Images/ImageEffect.cs

```diff
@@ -143,12 +148,79 @@ public void SetDepthOutput(Texture depthStencilView, params Texture[] renderTarg
         }
 
         /// <summary>
-        /// Sets the viewport to use .
+        /// Sets the viewport to use.
+        /// </summary>
+        /// <value>The viewport.</value>
+        public void SetViewport(Viewport? value)
+        {
+            if (!value.HasValue)
+            {
+                boundViewportCount = 0;
+            }
+            else
+            {
+                boundViewportCount = 1;
+                viewports[0] = value.Value;
+            }
+        }
+
+        /// <summary>
+        /// Sets the viewport to use.
         /// </summary>
-        /// <param name="viewport">The viewport.</param>
-        public void SetViewport(Viewport? viewport)
+        /// <param name="values">The viewports.</param>
+        public void SetViewports(Viewport[] values)
         {
-            this.viewport = viewport; // TODO: support multiple viewport?
+            SetViewports(values.Length, values);
+        }
+
+        /// <summary>
+        /// Sets the viewport to use.
+        /// </summary>
+        /// <param name="viewportCount">The viewports.</param>
+        /// <param name="values">The viewports.</param>
+        public void SetViewports(int viewportCount, Viewport[] values)
+        {
+            boundViewportCount = viewportCount;
+            if(viewportCount > 0)
+                Array.Copy(values, 0, this.viewports, 0, boundViewportCount);
```

Comment:
I think it would be preferable to do bounds check before calling copy so that you can throw a more meaningful ArgumentOutOfRangeException

Reply:
Array.Copy does the bound check internally and tells you that the arrays are not the same size.
If you take a look at CommandList we actually only have a for loop and get an IndexOutOfRange exception when you pass more than 16 elements.

Reply:
I know Array.Copy has bound checks, but they will be confusing to the user of this API unless they look into the source code when an exception is thrown. You need to check at least
* Count is not greater than array length
* Count is not greater than destination length